runners/java-fn-execution/src/main/java/org/apache/beam/runners/fnexecution/control/ProcessBundleDescriptors.java

```diff
@@ -373,6 +378,85 @@ private static TargetEncoding addStageOutput(
     return idsToSpec.build().rowMap();
   }
 
+  private static void sanitizeComponents(ExecutableStage stage, Builder components) {
```

Comment:
Are there other cases than removing dangling input from flatten?

Reply:
There are none at this time but in anycase we should sanitize the input to avoid dangling inputs.